addon/utils/check-model-for-dirty.js

```diff
@@ -12,7 +12,7 @@ function addDirtyModel(dirtyModels, model) {
   }
 
   model.forEach(m => {
-    if (m.get('isDirty')) {
+    if (m.get('hasDirtyAttributes') || m.get('isDirty')) {
```

Comment:
It could be a newer ember version, but `hasDirtyAttributes` could be false. Then it would check `isDirty` and warn/error.